calculator.c

```diff
@@ -0,0 +1,117 @@
+#include <stdio.h>
+#include <ctype.h>
+#include <stdlib.h>
+#define MAX_SIZE 100
+
+int operation(int a, int b, char op)
+{
+    switch (op)
+    {
+    case '+':
+        return a + b;
+    case '-':
+        return a - b;
+    case '*':
+        return a * b;
+    case '/':
+        if (b == 0)
+        {
+            printf("Error: Division by zero.\n");
+            exit(1);
+        }
+        return a / b;
+    default:
+        return 0;
+    }
+}
+
+int precedence(char op)
+{
+    if (op == '+' || op == '-')
+        return 1;
+    if (op == '*' || op == '/')
+        return 2;
+    return 0;
+}
+
+int evaluate(char *exp)
```

Comment:
Can you split this multiple function doing specific task as this function is doing multiple tasks?